src/Featurizers/Components/VectorNormsEstimator.h

```diff
@@ -0,0 +1,192 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+
+#include "TrainingOnlyEstimatorImpl.h"
+#include "../../Traits.h"
+#include "NormUpdaters.h"
+#include "../Structs.h"
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+static constexpr char const * const         VectorNormsEstimatorName("VectorNormsEstimator");
+
+
+template <typename T>
+struct IsIteratorPair {
+    static constexpr bool const value = false;
+};
+template <typename T>
+struct IsIteratorPair<std::tuple<T, T>> {
+    static constexpr bool const value = true;
+};
+template <typename T>
+struct IsIteratorPair<std::pair<T, T>> {
+    static constexpr bool const value = true;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         VectorNormsAnnotationData
+///  \brief         An annotation class which contains the norms of a matrix

```

Comment:
>An annotation class which contains the norms of a matrix [](start = 20, length = 56)

Nit: consider change this description to "An annotation class which contains the norms for all rows of a matrix."

(This was confusing to me at first, so maybe the clarification will be helpful to others)